engine/tests/vm38/test_phase_integration.py

```
@@ -0,0 +1,509 @@
+            frame = Frame(graph=g, cursor=a)
+            with pytest.raises(ValueError, match="validation failed"):
+                frame.follow_edge(edge)
+
+    def test_validate_successor_exists_blocks_dangling_anonymous_edge(self) -> None:
+        """An AnonymousEdge whose successor has no graph membership is blocked."""
+        g = Graph()
+        a = _node(g, label="a")
+        orphan = TraversableNode(label="orphan")  # NOT added to graph
+        edge = AnonymousEdge(predecessor=a, successor=orphan)
+        frame = Frame(graph=g, cursor=a)
+        with pytest.raises(Exception):
```

Ruff (0.15.1)

[warning] Do not assert blind exception: `Exception`

(B017)